serverless/aws/cdk/src/main/scala/sttp/tapir/serverless/aws/cdk/core/Request.scala

```diff
@@ -0,0 +1,27 @@
+package sttp.tapir.serverless.aws.cdk.core
+
+import sttp.model.Method
+import sttp.tapir.AnyEndpoint
+import sttp.tapir.EndpointInput.{FixedMethod, FixedPath, PathCapture}
+import sttp.tapir.internal.RichEndpoint
+
+private[core] case class Request(method: String, path: List[Segment]) {
+  def addPath(segment: Segment): Request = this.copy(path = path :+ segment)
+
+  def withMethod(m: Method): Request = this.copy(method = m.toString())
+}
+
+private[core] object Request {
+  def fromEndpoint(endpoint: AnyEndpoint): Request =
```

Comment:
I think here also `Endpoint.method` + `Endpoint.renderPathTemplate` might be useful

Reply:
I will definitely give it a try